OpenSim/Common/Component.h

```diff
@@ -712,6 +716,7 @@ OpenSim_DECLARE_ABSTRACT_OBJECT(Component, Object);
     */
     template <class C = Component>
     C& updComponent(const std::string& name) {
+        clearObjectIsUpToDateWithProperties();
```

Comment:
Yes, the `upd` methods should be reserved for Component (and Model) editing and not for simulation time querying. I think this makes `updComponent` (and I also made `udComponentList` comply) consistent to write access of `Set`s that are properties. For example, if you access bodies, forces, constraint via their Sets (e.g, `Model::updBodySet()`) these have the same effect of *dirtying* the up-to-date flag.